### ISSUE TITLE
Related fields should be able to reference a shared hint in their aria-describedby attribute

### ISSUE THREAD

**Issue by greatislander** [fluid-project/hearth#50]
## Is your feature request related to a problem?

Currently, Hearth provides some generic components which help with constructing form inputs:

- `<x-hearth-input>`
- `<x-hearth-textarea>`
- `<x-hearth-select>`

(Going forward, I'll use `<x-hearth-input>` as a generic example as this issue is shared by all three of these components.)

If an `<x-hearth-input>` has the `hinted` attribute, its `aria-describedby` attribute will be populated with a reference to an `id` for the hint in the form `<name>-hint` (where `<name>` is the `name` attribute of the input). The hint is generated using an `<x-hearth-hint>` component with a `for` attribute referencing the `name` / `id` of the input. Furthermore, if the input has failed validation, `aria-describedby` will also contain an `id` reference for an error with an `id` in the form of in the form `<name>-error` (where `<name>` is the `name` attribute of the input). Any errors can be generated using an `<x-hearth-error>` component with a `for` attribute referencing the `name` / `id` of the input and, optionally, a `bag` attribute referencing a [named error bag](https://laravel.com/docs/8.x/validation#named-error-bags). Here's a complete example:

```blade
<x-hearth-label for="fullname">Your name</x-hearth-label>
<x-hearth-input name="fullname" type="text" hinted />
<x-hearth-hint for="fullname">Enter your first and last name.</x-hearth-hint>
<x-hearth-error for="fullname" />
```

If the input is valid, this compiles to:

```html
<label for="name">Your name</label>
<input name="name" id="name" type="text" aria-describedby="name-hint">
<p class="field__hint" id="name-hint">Enter your first and last name.</p>
```

If the input is not valid, this compiles to:

```html
<label for="name">Your name</label>
<input name="name" id="name" type="text" aria-describedby="name-hint name-error" aria-invalid="true">
<p class="field__hint" id="name-hint">Enter your first and last name.</p>
<p class="field__error" id="name-error">This field is required.</p>
```

The problem arises when two related fields need to share a hint. For example, inputs for city/town and province/territory might have a shared hint which reads:

> Providing your location will allow us to show you local content.

There's currently no way to pass a reference to a specific hint to more than one input component.

## Describe the solution you'd like

The `hinted` attribute should accept an optional value equal to the `id` of the hint. So, in the example above, one could do the following:

```blade
<x-hearth-hint for="address">Providing your location will allow us to show you local content.</x-hearth-hint>
<x-hearth-label for="locality">City or town</x-hearth-label>
<x-hearth-input name="locality" type="text" hinted="address-hint" />
<x-hearth-error for="locality" />
<x-hearth-label for="region">Province or territory</x-hearth-label>
<x-hearth-input name="region" type="text" hinted="address-hint" />
<x-hearth-error for="region" />
```

This would compile to:

```html
<p class="field__hint" id="address-hint">Providing your location will allow us to show you local content.</p>
<label for="locality">City or town</label>
<input name="locality" id="locality" type="text" value="" aria-describedby="address-hint">
<label for="region">Province or territory</label>
<input name="region" id="region" type="text" value="" aria-describedby="address-hint">
```

## Describe alternative solutions you've considered

Not applicable.

## Additional context or notes

Not applicable.
